packages/wrangler/src/autoconfig/details.ts

```diff
@@ -118,23 +124,73 @@ export async function getDetailsForAutoConfig({
 		packageJson,
 		buildCommand: detectedFramework?.buildCommand ?? packageJsonBuild,
 		outputDir: detectedFramework?.dist ?? (await findAssetsDir(projectPath)),
+		workerName: toValidWorkerName(packageJson?.name ?? basename(projectPath)),
 	};
 }
 
+const invalidWorkerNameCharsRegex = /[^a-z0-9-]/g;
+const invalidWorkerNameStartEndRegex = /^-|-$/g;
+
+function checkWorkerNameValidity(
+	input: string
+): { valid: false; cause: string } | { valid: true } {
+	if (input.match(invalidWorkerNameStartEndRegex)) {
+		return {
+			valid: false,
+			cause: "Worker names cannot start or end with a dash.",
+		};
+	}
+
+	if (input.match(invalidWorkerNameCharsRegex)) {
+		return {
+			valid: false,
+			cause:
+				"Project names must only contain lowercase characters, numbers, and dashes.",
+		};
+	}
+
+	if (input.length > 58) {
+		return {
+			valid: false,
+			cause: "Project names must be less than 58 characters.",
+		};
+	}
+
+	return { valid: true };
+}
+
+function toValidWorkerName(name: string): string {
+	if (checkWorkerNameValidity(name).valid) {
+		return name;
+	}
+
+	name = name
+		// Replace all underscores with dashes
+		.replaceAll("_", "-")
+		// Remove all the special characters besides dashes
+		.replace(invalidWorkerNameCharsRegex, "")
```

Comment:
I think we should probably be replacing these with dashes rather than removing

Reply:
done 👍